daml-lf/data/src/test/scala/com/digitalasset/daml/lf/data/TimeSpec.scala

```diff
@@ -110,28 +104,185 @@ class TimeSpec extends AnyFreeSpec with Matchers with TableDrivenPropertyChecks
       an[IllegalArgumentException] should be thrownBy Timestamp.MaxValue.addMicros(1)
     }
 
-    "toString produces an ISO 8601 compliant string" in {
+    "toString produces a UTC ISO 8601 compliant string without nano seconds" in {
       val testCases = Table(
-        "timeStamp",
+        "timestamp",
         Timestamp.MinValue,
         Timestamp.Epoch,
-        Timestamp.assertFromString("1969-07-20T20:17:00Z"),
-        Timestamp.assertFromString("1969-07-20T20:17:00.1Z"),
-        Timestamp.assertFromString("1969-07-20T20:17:00.12Z"),
-        Timestamp.assertFromString("1969-07-20T20:17:00.123Z"),
-        Timestamp.assertFromString("1969-07-20T20:17:00.1234Z"),
-        Timestamp.assertFromString("1969-07-20T20:17:00.12345Z"),
-        Timestamp.assertFromString("1969-07-20T20:17:00.123456Z"),
+        Timestamp.assertStrictFromString("1969-07-20T20:17:00Z"),
+        Timestamp.assertStrictFromString("1969-07-20T20:17:00.1Z"),
+        Timestamp.assertStrictFromString("1969-07-20T20:17:00.12Z"),
+        Timestamp.assertStrictFromString("1969-07-20T20:17:00.123Z"),
+        Timestamp.assertStrictFromString("1969-07-20T20:17:00.1234Z"),
+        Timestamp.assertStrictFromString("1969-07-20T20:17:00.12345Z"),
+        Timestamp.assertStrictFromString("1969-07-20T20:17:00.123456Z"),
         Timestamp.MaxValue,
       )
-
-      forEvery(testCases) { date =>
-        val i = Instant.parse(date.toString)
+      forEvery(testCases) { timestamp =>
+        val i = Instant.parse(timestamp.toString)
         val micros =
           TimeUnit.SECONDS.toMicros(i.getEpochSecond) +
             TimeUnit.NANOSECONDS.toMicros(i.getNano.toLong)
 
-        micros shouldBe date.micros
+        micros shouldBe timestamp.micros
+      }
+    }
+
+    val fromStringFunctions = Table(
+      "function",
+      Timestamp.fromString(_),
+      Timestamp.strictFromString(_),
+      Timestamp.lenientFromString(_),
+    )
+
+    val lenientFromStringFunctions = Table(
+      "function",
+      Timestamp.fromString(_),
+      Timestamp.lenientFromString(_),
+    )
+
+    "fromString functions convert a UTC ISO 8601 compliant string without nano without loss of precision" in {
```

Comment:
Do we want to test the `+00:00` case mentioned in the ticket https://bugs.openjdk.org/browse/JDK-8166138 ?